### PR TITLE
Updated to support debugging for prebuilt extensions

### PR DESCRIPTION
**Pull request by 3coins**
Current cookiecutter doesn't enable front-end debugging for prebuilt extensions, as they have don't emit the correct source maps for the typescript files. Added post generate hook that updates and adds appropriate config required to generate source maps for prebuilt extension. Also, removed some redundant launch configuration and unused cookiecutter properties.

### Instructions to try this change
Make sure you have a installed pre-built extension setup.

Install cookiecutter.

```bash
pip install cookiecutter
```

Navigate to the root of your Jupyter project, then run

```bash
cookiecutter https://github.com/3coins/debug-config-cookiecutter.git --checkout fix-for-prebuilt-extensions
```

A number of prompts will pop up. DO NOT CHANGE `vscode_config_path`. Answer the rest of the prompts as appropriate for your project

Install the debug dev dependencies

```bash
jlpm install
```

Build the extension, this will generate the source maps for front-end debugging

```bash
jlpm build
```

Inside vscode, use the newly added "jlab debug" or "jlab debug (watch)" targets to start debugging.

### CODE REVIEW

**Comment by 3coins** [comment]
cc @ajbozarth @marthacryan

**Comment by 3coins** [comment]
@telamonian Can you please review and let me know if I can improve anything.

**Comment by marthacryan** [comment]
Tried this out on the auto-generated cookiecutter extension from here: https://github.com/jupyterlab/extension-cookiecutter-ts and it worked for all three types of extensions. However from the main JupyterLab repo and from the [elyra repo](https://github.com/elyra-ai/elyra) it produces this error: 
```
You've downloaded /Users/marthacryan/.cookiecutters/debug-config-cookiecutter before. Is it okay to delete and re-download it? [yes]: 
virtual_env_root []: 
chrome_bin_path []: 
has_envfile [n]: 
vscode_config_path [.vscode]: 
notebooks_dir [notebooks]: 
Traceback (most recent call last):
  File "/var/folders/4x/blshgydj44gb7d2hc9yshnkm0000gn/T/tmpba_6_t31.py", line 69, in <module>
    update_package_json()
  File "/var/folders/4x/blshgydj44gb7d2hc9yshnkm0000gn/T/tmpba_6_t31.py", line 19, in update_package_json
    if "webpackConfig" not in jlab_config:
TypeError: argument of type 'NoneType' is not iterable
ERROR: Stopping generation because post_gen_project hook script didn't exit successfully
Hook script failed (exit status: 1)
```

Running with the `-f` flag fixes this error

**Comment by 3coins** [comment]
@marthacryan Thanks for testing, this is a useful discovery.

Both JupyterLab and Elyra are multi-package repos, the cookiecutter in it's current form won't work for these projects as there are multiple packages that need to be setup for the front-end debugging to work. This is doable, but requires a more intelligent inspection of the workspace to determine single vs multi-package environments. 

Given that increased scope, I propose tackling multi-package projects in a separate PR.

**Review by marthacryan — APPROVED** [review]
As described above, I tested this PR and it worked for me. I don't know much about the cookiecutter end of things so I can't speak to the code as much but overall it looks good to me

File: {{cookiecutter.vscode_config_path}}/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+    module: {
+      rules: [
+        {
+          test: /\.js$/,
+          enforce: "pre",
+          use: ["source-map-loader"],
+        },
+      ],
```
Comment (fcollonval) [inline {{cookiecutter.vscode_config_path}}/webpack.config.js]: @3coins I think this one should be part of `@jupyterlab/builder`. Would you mind opening a PR for that?

Comment (3coins) [reply {{cookiecutter.vscode_config_path}}/webpack.config.js]: Sure, thanks for looking into this.

Comment (3coins) [reply {{cookiecutter.vscode_config_path}}/webpack.config.js]: Opened PR here.
https://github.com/jupyterlab/jupyterlab/pull/13765

**Pull request merged**